components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1194,35 +1273,10 @@ std::map<std::string, brave_rewards::BalanceReport> RewardsServiceImpl::GetAllBa
 void RewardsServiceImpl::GetCurrentBalanceReport() {
   ledger::BalanceReportInfo report;
   auto now = base::Time::Now();
-  bool success = ledger_->GetBalanceReport(
-      GetPublisherMonth(now),
-      GetPublisherYear(now),
-      &report);
-
+  bool success = ledger_->GetBalanceReport(GetPublisherMonth(now),
+                                           GetPublisherYear(now), &report);
   if (success) {
-    EventRouter* event_router = EventRouter::Get(profile_);
-    if (event_router) {
-      extensions::api::brave_rewards::OnCurrentReport::Properties properties;
-
-      properties.ads = report.earning_from_ads_;
-      properties.closing = report.closing_balance_;
-      properties.contribute = report.auto_contribute_;
-      properties.deposit = report.deposits_;
-      properties.grant = report.grants_;
-      properties.tips = report.one_time_donation_;
-      properties.opening = report.opening_balance_;
-      properties.total = report.total_;
-      properties.recurring = report.recurring_donation_;
-
-      std::unique_ptr<base::ListValue> args(
-          extensions::api::brave_rewards::OnCurrentReport::Create(properties)
-              .release());
-      std::unique_ptr<Event> event(
-          new Event(extensions::events::BRAVE_ON_CURRENT_REPORT,
-                    extensions::api::brave_rewards::OnCurrentReport::kEventName,
-                    std::move(args)));
-      event_router->BroadcastEvent(std::move(event));
-    }
+    TriggerOnGetCurrentBalanceReport(report);
```

Comment:
this is a separate issue from this PR, but shouldn't we notify the UI if this call fails?

Reply:
Agreed. cc: @NejcZdovc, let me know what you think.